test/mpi/test_alltoallv.cc

```diff
@@ -0,0 +1,150 @@
+                             ucc_test_team_t &_team,
+                             ucc_test_vsize_flag_t _count_bits,
+                             ucc_test_vsize_flag_t _displ_bits) :
+    TestCase(_team, _mt, _msgsize, _inplace)
+{
+    size_t dt_size = ucc_dt_size(_dt);
+    size_t count = _msgsize/dt_size;
+    int rank;
+    int nprocs;
+    int rank_count;
+
+    std::default_random_engine eng;
+    eng.seed(test_rand_seed);
+    std::uniform_int_distribution<int> urd(count/2, count);
+
+    MPI_Comm_rank(team.comm, &rank);
+    MPI_Comm_size(team.comm, &nprocs);
+
+    dt = _dt;
+    sncounts = 0;
+    rncounts = 0;
+    scounts = NULL;
+    sdispls = NULL;
+    rcounts = NULL;
+    rdispls = NULL;
+    count_bits = _count_bits;
+    displ_bits = _displ_bits;
+
+    MPI_Comm_rank(team.comm, &rank);
+    MPI_Comm_size(team.comm, &nprocs);
+
+    args.coll_type = UCC_COLL_TYPE_ALLTOALLV;
+
+    if (count_bits == TEST_FLAG_VSIZE_64BIT) {
+        args.mask |= UCC_COLL_ARGS_FIELD_FLAGS;
+        args.flags |= UCC_COLL_ARGS_FLAG_COUNT_64BIT;
+    }
+    if (displ_bits == TEST_FLAG_VSIZE_64BIT) {
+        args.mask |= UCC_COLL_ARGS_FIELD_FLAGS;
+        args.flags |= UCC_COLL_ARGS_FLAG_DISPLACEMENTS_64BIT;
+    }
+
+    scounts = (int*)malloc(sizeof(*scounts) * nprocs);
+    sdispls = (int*)malloc(sizeof(*sdispls) * nprocs);
+    rcounts = (int*)malloc(sizeof(*rcounts) * nprocs);
+    rdispls = (int*)malloc(sizeof(*rdispls) * nprocs);
+
+    for (auto i = 0; i < nprocs; i++) {
+        rank_count = urd(eng);
+        scounts[i] = rank_count;
+    }
+
+    MPI_Alltoall((void*)scounts, 1, MPI_INT,
+                 (void*)rcounts, 1, MPI_INT, team.comm);
+
+    sncounts = 0;
+    rncounts = 0;
+    for (auto i = 0; i < nprocs; i++) {
+        assert((size_t)rcounts[i] <= count);
+        sdispls[i] = sncounts;
+        rdispls[i] = rncounts;
+        sncounts += scounts[i];
+        rncounts += rcounts[i];
+    }
+
+    if (skip((test_max_size &&
+              (test_max_size < (sncounts * dt_size)
+               || test_max_size < (rncounts * dt_size))),
+        TEST_SKIP_MEM_LIMIT, team.comm)) {
+        return;
+    }
+
+    UCC_CHECK(ucc_mc_alloc(&sbuf, sncounts * dt_size, _mt));
+    UCC_CHECK(ucc_mc_alloc(&check_sbuf, sncounts * dt_size,
+                           UCC_MEMORY_TYPE_HOST));
+    UCC_CHECK(ucc_mc_alloc(&rbuf, rncounts * dt_size, _mt));
+    UCC_CHECK(ucc_mc_alloc(&check_rbuf, rncounts * dt_size,
+                           UCC_MEMORY_TYPE_HOST));
+    init_buffer(sbuf, sncounts, dt, _mt, rank);
+    init_buffer(check_sbuf, sncounts, dt, UCC_MEMORY_TYPE_HOST, rank);
+
+    args.src.info_v.buffer = sbuf;
+    args.src.info_v.datatype = _dt;
+    args.src.info_v.mem_type = _mt;
+
+    args.dst.info_v.buffer = rbuf;
+    args.dst.info_v.datatype = _dt;
+    args.dst.info_v.mem_type = _mt;
+
+    if (TEST_FLAG_VSIZE_64BIT == count_bits) {
+        args.src.info_v.counts =
```

Comment:
_mpi_count_to_ucc  allocates memory,  where do you call free() for it?

Reply:
fixed